doc/introduction.md

```diff
@@ -1,16 +1,86 @@
 # Introduction
 
-RxJS is a library for composing asynchronous and event-based programs by using observable sequences. It provides one core type, the [Observable](./overview.html#observable), satellite types (Observer, Schedulers, Subjects) and operators inspired by [Array#extras](https://developer.mozilla.org/en-US/docs/Web/JavaScript/New_in_JavaScript/1.6) (map, filter, reduce, every, etc) to allow handling asynchronous events as collections.
-
```

Comment:
I do agree with your reasoning that very early on people should be exposed to examples on what RxJS does in actual real-world code, like with the examples you added. But I do not at all agree with removing all the short explanations that this PR removes. There are different styles of learning, not everyone wants to jump into examples before they have even a clue "what is this about". `introduction.md` keeps it really short, it's just half a screen, we definitely need to keep it. I'd suggest you get these new examples and put them under a section `## First examples` at the bottom of this `introduction.md`.